tests/test_pre_processing.py

```diff
@@ -347,4 +347,162 @@ def test_no_annuity_no_cost_params_logs_message(self, caplog):
+                    os.path.join(self.pre_p_dir, "datapackage.json"))
+        # call the pre_processing function with wacc = 1 and custom_attributes list defined
+        wacc = 1
+        pre_processing(self.pre_p_dir, wacc=wacc,
+                       custom_attributes=["emission_factor", "renewable_factor", "land_requirement"])
+        # read the updated csv file
+        updated_df = pd.read_csv(os.path.join(self.package_path, f_name), sep=';')
+        print(updated_df.columns)
+        # assert that 'output_parameters' column is in the updated dataframe
+        assert 'output_parameters' in updated_df.columns, "'output_parameters' column is not present " \
+                                                          "in the updated dataframe"
+
+    def test_output_params_not_added_to_csv_with_cust_attr_in_csv_and_not_list(self):
+        """
+        Tests that the output parameters column has not been added to the csv file if the csv
+        file contains custom attributes and they have not been defined as list by user.
+        """
+        # copy scenario csv file and datapackage json file to the package path
+        f_name = "cust_attr.csv"
+        dp_name = "dp_no_output_params.json"
+        shutil.copy(os.path.join(self.test_inputs_pre_p, f_name), os.path.join(self.package_path, f_name))
+        shutil.copy(os.path.join(self.test_inputs_pre_p, dp_name),
+                    os.path.join(self.pre_p_dir, "datapackage.json"))
+        # read the original dataframe before pre-processing
+        original_df = pd.read_csv(os.path.join(self.package_path, f_name), sep=';')
+        # call the pre_processing function with wacc = 1 and custom_attributes = none (default)
+        wacc = 1
+        pre_processing(self.pre_p_dir, wacc=wacc)
+        # read the updated csv file after pre-processing
+        updated_df = pd.read_csv(os.path.join(self.package_path, f_name), sep=';')
+        # assert that 'output_parameters' column is not in the updated dataframe if it wasn't present initially
+        if 'output_parameters' not in original_df.columns:
+            assert 'output_parameters' not in updated_df.columns, \
+                "'output_parameters' has been added to the updated dataframe when it should not be"
+
+    def test_output_params_added_to_json_with_cust_attr_in_csv_and_list(self):
+        """
+        Tests that the output parameters field has been added to the datapackage.json file if the
+        csv file contains custom attributes and they have been defined as list by user.
+        """
+        # copy scenario csv file and datapackage json file to the package path
+        f_name = "cust_attr.csv"
+        dp_name = "dp_no_output_params.json"
+        shutil.copy(os.path.join(self.test_inputs_pre_p, f_name), os.path.join(self.package_path, f_name))
+        shutil.copy(os.path.join(self.test_inputs_pre_p, dp_name),
+                    os.path.join(self.pre_p_dir, "datapackage.json"))
```

Comment:
This code is redundant, I would move it within a method which takes f_name and dp_name as arguments